Source/Core/Core/ActionReplay.cpp

```diff
@@ -303,6 +304,24 @@ bool IsSelfLogging()
 	return logSelf;
 }
 
+static void HostWrite_U8(const u8 var, const u32 address)
+{
+	PowerPC::HostWrite_U8(var, address);
+	JitInterface::InvalidateICache(address, 1, false);
```

Comment:
PowerPC instructions are all 32bit in size and aligned to 32bit.  I think one of the reasons why the change takes 5 seconds to take effect is because the PR is not invalidating 32bits at a time.  Try changing this line and all of the other InvalidateICache calls to:

JitInterface::InvalidateICache(address & ~3, 4, true);

The forced flag should probably be set as the AR code might be changing a FIFO write call.